app/models/advisory_lock.rb

```diff
@@ -0,0 +1,19 @@
+class AdvisoryLock
```

Comment:
Should we call this `SIPTrunkChannelManager`? Because `AdvisoryLock` is just a concept of an implementation detail, so it shouldn't be tight coupling with our domain.